src/ngx_http_lua_ssl.c

```diff
@@ -34,4 +39,135 @@ ngx_http_lua_ssl_init(ngx_log_t *log)
 }
 
 
+#ifndef NGX_LUA_NO_FFI_API
+
+
+void *
+ngx_http_lua_ffi_ssl_ctx_init(ngx_uint_t protocols, char **err)
+{
+    SSL_CTX                 *ssl_ctx;
+    ngx_ssl_t                ssl;
+
+    ssl.log = ngx_cycle->log;
+    if (ngx_ssl_create(&ssl, protocols, NULL) != NGX_OK) {
+        *err = "failed to create ssl ctx";
+        ngx_log_error(NGX_LOG_ERR, ssl.log, 0, *err);
+        return NULL;
+    }
+
+    ssl_ctx = ssl.ctx;
+
+    ngx_log_debug2(NGX_LOG_DEBUG_HTTP, ngx_cycle->log, 0,
+                   "lua ssl ctx init: %p:%d", ssl_ctx, ssl_ctx->references);
+
+    return ssl_ctx;
+}
+
+
+int
+ngx_http_lua_ffi_ssl_ctx_set_cert(void *cdata_ctx, void *cdata_cert, char **err)
+{
+#ifdef LIBRESSL_VERSION_NUMBER
+
+    *err = "LibreSSL not supported";
+    return NGX_ERROR;
+
+#else
+
+#   if OPENSSL_VERSION_NUMBER < 0x1000205fL
+
+    *err = "at least OpenSSL 1.0.2e required but found " OPENSSL_VERSION_TEXT;
```

Comment:
The API [SSL_CTX_add1_chain_cert](https://www.openssl.org/docs/man1.1.0/ssl/SSL_CTX_add1_chain_cert.html) says 
> These functions were first added to OpenSSL 1.0.2.

So I choose to requiring 1.0.2e in order to keep consistent with other function requires